btree/btree.c

```diff
@@ -3179,6 +3192,41 @@ static bool address_in_segment(struct segaddr addr)
 	return true;
 }
 
+/**
+ * Callback to be invoked on transaction commit.
+ */
+static void btree_tx_commit_cb(void *payload)
+{
+	struct nd *node = payload;
+
+	node_lock(node);
+	M0_ASSERT(node->n_txref != 0);
+	node->n_txref--;
+	node_unlock(node);
+}
+
+static void btree_tx_node_capture(struct m0_btree_oimpl *oi,
+				  struct m0_be_tx *tx)
+{
+	struct node_capture_info *arr = oi->i_capture;
+	/* struct slot          node_slot; */
+	int                       i;
+
+	for (i = 0; i < BTREE_CALLBACK_CREDIT; i++) {
+		if (arr[i].nc_node == NULL)
+			break;
+		/**
+		 * node_slot.s_node = arr[i].nc_node;
+		 * node_slot.s_idx  = arr[i].nc_idx;
+		 * node_capture(&node_slot, bop->bo_tx);
+		 */
+		node_lock(arr[i].nc_node);
+		arr[i].nc_node->n_txref++;
+		node_unlock(arr[i].nc_node);
```

Comment:
Is it required to take node_lock , as we are performing  btree_tx_commit_cb and btree_tx_node_capture*(which access nc_node->n_txref ) in tree_lock , So no other thread will be accessing nc_node->n_txref unless there are some other functions which can access nc_node->n_txref .

Reply:
btree_tx_commit_cb will be invoked on tx_commit, which will be in tree_unlock mode.